trl/trainer/ppov2_trainer.py

```diff
@@ -220,7 +220,7 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = Fa
             self.model = self.accelerator.unwrap_model(self.model).policy  # save only the policy
         if output_dir is None:
             output_dir = self.args.output_dir
-        state_dict = self.accelerator.get_state_dict(self.backup_model)
+        state_dict = self.accelerator.get_state_dict(self.model)
```

Comment:
This is prob incorrect. There are two scenarios:

1. We call `trainer.save_model`, in which case, `if not _internal_call` gets triggered, and `self.model` becomes the policy
2. we call `trainer.push_to_hub`, in which case, `push_to_hub` sets the `self.model` to be the policy, and `super().push_to_hub(**kwargs)` calls `save_model(..., _internal_call=True)`, and in that case `self.model` is still the policy.

Reply:
But how is `self.backup_model` set in the case when `_internal_call`. I don't see it being set and I'm getting an error